.github/workflows/lint-golang.yml

```diff
@@ -13,7 +13,7 @@ on:
       - 'src/go/**'
 
 env:
-  GOPRIVATE: github.com/redpanda-data/flux-controller-shim
+  GOPRIVATE: github.com/redpanda-data/*
```

Comment:
NIT: I personally would like to be explicit with private dependencies. 

P.S.
To add more I don't know why public project has private dependency. None of the community contributor could compile our operator.